### PR TITLE
[fix] Fix name of self-CA in certificate manager

### PR DESCRIPTION
**Pull request by alexAubin**
Fixes https://dev.yunohost.org/issues/652

### CODE REVIEW

**Comment by opi** [comment]
I'd like other people to test this PR by generating self-signed certificate, but as it is a bugfix, I don't wan't to wait 2 weeks before merging. This need to be shipped in the next stable release, along LE integration work.

File: src/yunohost/certificate.py [hunk 1]
```diff
@@ -810,9 +810,20 @@ def _domain_is_accessible_through_HTTP(ip, domain):
...
+    ca_conf = os.path.join(SSL_DIR, "openssl.ca.cnf")
+
+    try :
+        with open(ca_conf) as f:
+            lines = f.readlines()
+
+        for line in lines:
+            if line.startswith("commonName_default"):
+                return line.split()[2]
+    except:
```
Comment (Psycojoker) [inline src/yunohost/certificate.py]: We should never silencely hide exception, this is always a debugging nightmare, we should either handle the case we think that could fail in the code and/or printing warning on exception catach

Comment (alexAubin) [inline src/yunohost/certificate.py]: My try/except was aiming to catch the case where ca_conf doesn't exist, essentially... and to not block the rest of the program if something bad happens (hence the `logger.warning`). Would you like me to replace the try/except by something like `if os.path.exists(ca_conf)` ?

Comment (Psycojoker) [inline src/yunohost/certificate.py]: You can do both and print the exception + the stack if something bad happen, but never silencly hide exception, this is really really bad for debugging :/

File: src/yunohost/certificate.py [hunk 2]
```diff
@@ -810,9 +810,20 @@ def _domain_is_accessible_through_HTTP(ip, domain):
 
 
 def _name_self_CA():
-    cert = crypto.load_certificate(
-        crypto.FILETYPE_PEM, open(SELF_CA_FILE).read())
-    return cert.get_subject().CN
+    ca_conf = os.path.join(SSL_DIR, "openssl.ca.cnf")
```
Comment (julienmalik) [inline src/yunohost/certificate.py]: Seems a little wrong to me to get the info you're looking for from this file.
Shouldn't it be interpreted from the cert itself ?

Is it what you are looking for ? :
```
openssl  x509 -text -noout -in /usr/share/yunohost/yunohost-config/ssl/yunoCA/ca/cacert.pem
```
then grep for ```Subject : CN=domain.com```

Comment (julienmalik) [inline src/yunohost/certificate.py]: There is a ```Subject : CN=``` and a ```Issuer : CN=``` whenprinting the cert content.
So if both domains are the same, then it is self-signed ?

Comment (julienmalik) [inline src/yunohost/certificate.py]: maybe doable with this crypto lib, I don't know.

Comment (alexAubin) [inline src/yunohost/certificate.py]: This is what was done before, actually... But for some reason, opi had a setup where he changed the `cacert.pem` with something else. The "true" info is coming from the `openssl.ca.cnf`, from what I investigated (could be wrong, looks like there's some obscure magic going on in openssl).

Comment (opi) [inline src/yunohost/certificate.py]: maybe we can forget about my strange setup ? ( bit of history: I used to have sefl-signed certificate, with CA on main domain; then change the main domain, then dirty hack to set up a gandi certificate, and then hack with letsencrypt by hand ).

File: src/yunohost/certificate.py (hunk 1 above)
Comment (alexAubin) [inline src/yunohost/certificate.py]: Should be fixed in 6aa64a0

**Pull request merged**